rust/lance-table/src/rowids/segment.rs

```diff
@@ -273,49 +273,119 @@ impl U64Segment {
+    //         }
+    //         Self::RangeWithBitmap { range, bitmap } => {
+    //             // We need to find the actual position in the bitmap for the given offset
+    //             // taking cleared bits (holes) into account
+    //             let mut actual_start_pos = None;
+    //             let mut actual_end_pos = None;
+    //             let mut position = 0;
+    //
+    //             for (i, bit) in bitmap.iter().enumerate() {
+    //                 if bit {
+    //                     if position == offset {
+    //                         actual_start_pos = Some(i);
+    //                     }
+    //                     if position == offset + len {
+    //                         actual_end_pos = Some(i);
+    //                         break;
+    //                     }
+    //                     position += 1;
+    //                 }
+    //             }
+    //
+    //             // Handle case where we reach the end
+    //             if actual_end_pos.is_none() && position == offset + len {
+    //                 actual_end_pos = Some(bitmap.len());
+    //             }
+    //
+    //             let start_pos = actual_start_pos.unwrap_or(0);
+    //             let end_pos = actual_end_pos.unwrap_or(bitmap.len());
+    //
+    //             let bitmap_len = end_pos - start_pos;
+    //             let new_bitmap = bitmap.slice(start_pos, bitmap_len);
+    //             let new_range = (range.start + start_pos as u64)..(range.start + end_pos as u64);
+    //
+    //             if new_bitmap.count_ones() == bitmap_len {
+    //                 // All bits are set, so we can simplify to just a Range
+    //                 Self::Range(new_range)
+    //             } else {
+    //                 Self::RangeWithBitmap {
+    //                     range: new_range,
+    //                     bitmap: new_bitmap.into(),
+    //                 }
+    //             }
+    //         }
+    //         Self::SortedArray(array) => Self::SortedArray(array.slice(offset, len)),
+    //         Self::Array(array) => Self::Array(array.slice(offset, len)),
+    //     }
+    // }
+
     pub fn slice(&self, offset: usize, len: usize) -> Self {
-        match self {
-            Self::Range(range) => {
-                let start = range.start + offset as u64;
-                Self::Range(start..(start + len as u64))
-            }
-            Self::RangeWithHoles { range, holes } => {
-                let start = range.start + offset as u64;
-                let end = start + len as u64;
+        if len == 0 {
+            return Self::Range(0..0);
+        }
 
-                let start = holes.binary_search(start).unwrap_or_else(|x| x) as u64;
-                let end = holes.binary_search(end).unwrap_or_else(|x| x) as u64;
-                let holes_len = end - start;
+        let slice_iter = self.iter().skip(offset).take(len);
+        
+        let values: Vec<u64> = slice_iter.collect();
```

Comment:
nit: why not just `let values = self.iter().skip(offset).take(len).collect<Vec<_>>()`